pagerduty.tf

```diff
@@ -1,24 +1,118 @@
-resource "newrelic_alert_channel" "pagerduty" {
+locals {
+  summary       = "{{ annotations.title.[0] }}"
+  customDetails = <<-EOT
+    {
+      "condition_name": {{json accumulations.conditionName.[0]}},
+      "details": {{json issueTitle}},
+      "duration": {{#if issueDurationMs}}{{issueDurationMs}}{{else}}0{{/if}},
+      "event_type": "INCIDENT",
+      "incident_acknowledge_url": {{json issueAckUrl}},
+      "incident_url": {{json issuePageUrl}},
+      "policy_name": {{json accumulations.policyName.[0]}},
+      {{#if accumulations.runbookUrl}}"runbook_url": {{json accumulations.runbookUrl.[0]}},{{/if}}
+      "severity": "{{#eq 'HIGH' priority}}WARNING{{else}}{{priority}}{{/eq}}",
+      "status": {{json status}},
+      "timestamp": {{updatedAt}},
+      "timestamp_utc_string": {{json issueUpdatedAt}}
+    }
+    EOT
+}
+
+resource "newrelic_notification_destination" "pagerduty" {
   count = var.enable_pagerduty_notifications ? 1 : 0
 
-  name = var.newrelic_app_name
-  type = "pagerduty"
+  account_id = var.newrelic_account_id
+  name       = var.newrelic_app_name
+  type       = "PAGERDUTY_SERVICE_INTEGRATION"
 
-  config {
-    service_key = var.pagerduty_service_key
+
+  auth_token {
+    prefix = "Token token="
+    token  = var.pagerduty_service_key
+  }
+
+  property {
+    key   = "two_way_integration"
+    value = "false"
   }
 }
 
-resource "newrelic_alert_policy_channel" "pagerduty_non_urgent" {
+resource "newrelic_notification_channel" "pagerduty_urgent_channel" {
   count = var.enable_pagerduty_notifications ? 1 : 0
 
-  policy_id   = newrelic_alert_policy.non_urgent.id
-  channel_ids = [newrelic_alert_channel.pagerduty[0].id]
+  account_id     = var.newrelic_account_id
+  name           = var.newrelic_app_name
+  type           = "PAGERDUTY_SERVICE_INTEGRATION"
+  destination_id = newrelic_notification_destination.pagerduty[0].id
+  product        = "IINT"
+
+  property {
+    key   = "summary"
+    value = local.summary
+  }
+
+  property {
+    key   = "customDetails"
+    value = local.customDetails
+  }
 }
 
-resource "newrelic_alert_policy_channel" "pagerduty_urgent" {
+resource "newrelic_notification_channel" "pagerduty_non_urgent_channel" {
   count = var.enable_pagerduty_notifications ? 1 : 0
 
-  policy_id   = newrelic_alert_policy.urgent.id
-  channel_ids = [newrelic_alert_channel.pagerduty[0].id]
+  account_id     = var.newrelic_account_id
+  name           = var.newrelic_app_name
+  type           = "PAGERDUTY_SERVICE_INTEGRATION"
+  destination_id = newrelic_notification_destination.pagerduty[0].id
+  product        = "IINT"
+
+  property {
+    key   = "summary"
+    value = "{{ annotations.title.[0] }}"
```

Comment:
Why is the summary different compared to the one used in `pagerduty_urgent_channel`?

Reply:
I forgot to update it 🤦 .